ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java

```diff
@@ -2994,12 +3005,13 @@ Seems much cleaner if each stmt is identified as a particular HiveOperation (whi
         break;
       case DDL_SHARED:
         compBuilder.setSharedRead();
-        if (!output.isTxnAnalyze()) {
+        if (output.isTxnAnalyze()) {
           // Analyze needs txn components to be present, otherwise an aborted analyze write ID
           // might be rolled under the watermark by compactor while stats written by it are
           // still present.
-          compBuilder.setOperationType(DataOperationType.NO_TXN);
+          continue;
```

Comment:
With using 'continue' we do not execute the code after the case statement. Is this intentional?

Reply:
analyze wasn't adding new locks even before, it tried to add duplicate on table level and failed as there was already one. when i removed table level lock addition, analyze started to fail as it doesn't even populate operation type.